projects/hg-resolvers/src/lib/resolver.ts

```diff
@@ -278,7 +278,7 @@ export class Resolver<T, D = any> {
 
     return !dependencies ? of(undefined) : combineLatest(...dependencies).pipe(
       (isAutoResolveOnceConfig || isDefaultConfig) ? first() : takeUntil(this._isAlive$),
-      filter(() => !this.shouldSkip)
+      filter(() => !this.shouldSkip && this.state !== ResolverState.PENDING)
```

Comment:
Why do we need the extra check for state === PENDING? What if we are waiting for something to get resolved and it's taking a while but meanwhile the dependencies change. That means that we don't care anymore for the previous request and we should cancel it and dispatch the new one.

Reply:
Leftover from the debugging process :D , my bad.